components/search/src/lib.rs

```diff
@@ -36,17 +38,57 @@ fn build_fields(search_config: &Search) -> Vec<String> {
         fields.push("description".to_owned());
     }
 
+    if search_config.include_path {
+        fields.push("path".to_owned());
+    }
+
     if search_config.include_content {
         fields.push("body".to_owned());
     }
 
     fields
 }
 
+fn path_tokenizer(text: &str) -> Vec<String> {
+    text.split(|c: char| c.is_whitespace() || c == '-' || c == '/')
+        .filter(|s| !s.is_empty())
+        .map(|s| s.trim().to_lowercase())
+        .collect()
+}
+
+fn build_tokenizers(search_config: &Search, language: Language) -> Vec<TokenizerFn> {
+    let text_tokenizer = match language {
+        #[cfg(feature = "indexing-zh")]
+        Language::Chinese => pipeline::tokenize_chinese,
+        #[cfg(feature = "indexing-ja")]
+        Language::Japanese => pipeline::tokenize_japanese,
+        _ => pipeline::tokenize,
+    };
+    let mut tokenizers: Vec<TokenizerFn> = vec![];
+    if search_config.include_title {
+        tokenizers.push(text_tokenizer);
+    }
+
+    if search_config.include_description {
+        tokenizers.push(text_tokenizer);
+    }
+
+    if search_config.include_path {
+        tokenizers.push(path_tokenizer);
+    }
+
+    if search_config.include_content {
+        tokenizers.push(text_tokenizer);
+    }
+
+    tokenizers
```

Comment:
You can end up with the same tokenizer multiple times, is that not an issue?

Reply:
The same tokenizer, that is the default tokenizer, should be used for all fields except the path field which gets it's own tokenizer that splits on `/`. Those are just function pointers so having two pointers point to the same function is not an issue.